src/Service/ProgrammesService.php

```diff
@@ -197,6 +197,38 @@ function () use ($pid, $entityType) {
         );
     }
 
+    /**
+     * @param Pid[] $pids
+     * @param string $entityType
+     * @param string $ttl
+     * @return Programme[]
+     */
+    public function findByPids(array $pids, string $entityType = 'Programme', $ttl = CacheInterface::NORMAL)
+    {
+        $this->assertEntityType($entityType, self::ALL_VALID_ENTITY_TYPES);
+
+        $validPids = [];
+
+        foreach ($pids as $pid) {
+            if ($pid instanceof Pid) {
```

Comment:
Do we assert an array only contains instances of something any where else in PPS? I feel this should be genericied.

Reply:
A few places yeah. I believe you raised a ticket to look into an Assert library that offers these generic checks

Reply:
Did I? Well done past Mike.

Reply:
Ah no, I lied, I think you mentioned it to me then I raised https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5848